docker/scripts/Test-AzureStack.ps1

```diff
@@ -1,75 +1,309 @@
 # Sample script to execute a set of basic tests using the Admin and User Az modules aganist provided AzureStack Environment 
-# First setup Azure PowerShell context by running Initialize-AzurePowerShellEnvironment.ps1
+# First setup Azure PowerShell context by running Login-Environment.ps1 and New-AzureStackTestResources.ps1.
+
 [CmdletBinding()]
 param (
+    [Parameter(Mandatory=$true)]
+    [string]
+    $ClientObjectId = $ENV:ClientObjectId
 )
 
+$resourceGroup = "azurestack-container-rg"
+$resourceLocation = "northwest"
+$globalLogFile = "log-" + (Get-Date).ticks + ".txt"
+
 [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12 
 
-function Log([System.Object] $param)
+filter timestamp 
 {
-    Write-Verbose -Message "$([datetime]::UtcNow.ToString("o")): $($param | Out-String)" -Verbose
+    "[$((Get-Date).ToUniversalTime().toString("o"))] $_"
 }
 
-function Setup 
+function Log
 {
-  Log -param "START: $($MyInvocation.MyCommand)"
-
-  $context = Get-AzContext
+    Param
+    (
+        [Parameter(ValueFromPipeline = $true)]
+        [String]
+        $message,
+        [ValidateSet("INFO", "EXCEPTION")]
+        [String]
+        $type = "INFO"
+    )
 
-  if(-not $context)
-  {
-    $msg = "First setup AzureStack context by running Initialize-AzurePowerShellEnvironment.ps1" 
-    Log -param $msg
-    Write-Error -Message $msg
-  }
-  
-  Log -param "Subscription List:"
-  Get-AzSubscription
+    if ($type -like "INFO")
+    {
+        $message = "[INFO] " + $message
+    }
+    elseif ($type -like "EXCEPTION")
+    {
+        $message = "[EXCEPTION] " + $message
+    }
 
-  Log -param "END: $($MyInvocation.MyCommand)"
+    $message | timestamp | Tee-Object -FilePath $globalLogFile -Append | Write-Host
 }
 
-function Get-ArmEndpoint
+function Test
 {
-    $context = Get-AzContext
-    $ARMEndpoint = $context.Environment.ResourceManagerUrl
-    return $ARMEndpoint
-}
+    Param
+    (
+        [String] $testFunctionName,
```

Comment:
convention is to use Pascal casing for parameter naming , testFunctionName -> TestFunctionName